pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -66,7 +66,8 @@ func init() {
 }
 
 var (
-	azDebug = os.Getenv("VERBOSE_AZ_DEBUG") == "1"
+	azDebug   = os.Getenv("VERBOSE_AZ_DEBUG") == "1"
+	directEDS = os.Getenv("PILOT_DIRECT_EDS") != "0"
```

Comment:
We must have a flag or env or config for each new feature AFAIK, so in case of problems we can turn it off. Either way - there will be as many using configs or flags, but will be harder to cleanup - the linter will complain if one of them is unused. 

With mesh config or flags things are much harder on both add and remove - changes in many files ( values, template, main, model, passing it around, finally the object itself). And upgrade may become an issue - like we saw with Nate's readiness probe. 

We set only a few - ideally zero. They are used for fail-safe, if in a certain environment things 
are very different.

I will add more comments and make them 'exported' so godoc captures them. And in master I can get rid of many of them. If something really needs to be set by user - will move into proper 
config.

Reply:
Werent these moved to feature flags file? pkg/features/pilot/...